docs/data/base/components/snackbar/snackbar.md

```diff
@@ -0,0 +1,145 @@
+
+## Introduction
+
+A snackbar provides users with a brief, temporary message about app processes without interrupting their activity or experience.
+
+The `SnackbarUnstyled` component is built to appear in the corner or center of the screen to inform users about an action that the app is taking.
+
+{{"component": "modules/components/ComponentLinkHeader.js", "design": false}}
+
+## Component
+
+### Usage
+
+After [installation](/base/getting-started/installation/), you can start building with this component using the following basic elements:
+
+```jsx
+import SnackbarUnstyled from '@mui/base/SnackbarUnstyled';
+
+export default function MyApp() {
+  return <SnackbarUnstyled>{/* snackbar text */}</SnackbarUnstyled>;
+}
+```
+
+### Basics
+
+:::info
+Snackbars are differentiated from other kinds of popups and alerts in that they are _informational, temporary, and contextual_ (Source: [Material Design—Snackbars](https://material.io/components/snackbars)).
+:::
+
+`SnackbarUnstyled` doesn't impose any restrictions on its implementation—it's up to you to design it so that it doesn't interrupt the user experience, and disappears after a set amount of time without requiring the user to take action.
+
+Use the `autoHideDuration` prop to set the time (in milliseconds) that the snackbar remains on the screen.
+
+:::info
+You may want to implement `SnackbarUnstyled` with [`ClickAwayListener`](/base/react-click-away-listener/), so that the user can choose to dismiss the snackbar before its time is up by clicking anywhere outside of it.
+But this behavior is optional for a snackbar.
+:::
+
+The following demo illustrates the basic usage of `SnackbarUnstyled`.
+Click **Open snackbar** to see how it behaves:
+
+{{"demo": "UnstyledSnackbar.js", "defaultCodeOpen": false}}
+
+### Anatomy
+
+The `SnackbarUnstyled` component is composed of a single root `<div>` slot with no interior slots:
+
+```html
+<div role="presentation" className="BaseSnackbar-root">snackbar content</div>
+```
+
+### Slot props
+
+:::info
+The following props are available on all non-utility Base components.
+See [Usage](/base/getting-started/usage/) for full details.
+:::
+
+Use the `component` prop to override the root slot with a custom element:
+
+```jsx
+<SnackbarUnstyled component="span" />
+```
+
+Use the `components` prop to override any interior slots in addition to the root:
+
+```jsx
+<SnackbarUnstyled components={{ Root: 'span' }} />
+```
+
+:::warning
+If the root element is customized with both the `component` and `components` props, then `component` will take precedence.
+:::
+
+Use the `componentsProps` prop to pass custom props to internal slots.
+The following code snippet applies a CSS class called `my-snackbar` to the root slot:
+
+```jsx
+<SnackbarUnstyled componentsProps={{ root: { className: 'my-snackbar' } }} />
+```
+
+:::warning
+Note that `componentsProps` slot names are written in lowercase (`root`) while `components` slot names are capitalized (`Root`).
+:::
```

Comment:
I don't think this section provides any value for the deverloper as the component doesn't have additional internal slots. cc @samuelsycamore should we stip this from the components that support only the root slot?

Reply:
Regarding `componentsProps`, we also have `componentProps.clickAwayListener` in this component. We can remove the `components` prop part as there is only `Root` slot.

Reply:
Hmm that's a good question @mnajdova. I like the uniformity of including this information in all of the component docs, but you're right that it is overkill for a component like this one. Is there a use case where a developer would want/need to use `components` here instead of `component`? My only concern would be that devs might think you _can't_ use `components` on a component like this one, even though you can (but probably don't need to).

Reply:
Let's leave it as is for now, until we decide on the `components` -> `slots` rename, as well as what will the name of the `component` prop be in MUI Base. We can revisit it later.

To answer your question @samuelsycamore there is no difference in MUI Base between `component` vs components.Root`, the first one is a shorthand. However, this would behave a bit differently in Material UI and Joy UI, because there the `component` prop acts like the emotion's "as" prop, which means it will only change the render HTML element, instead of replacing everything with the new component (for example, the styles and everything else created in the component will still apply. This is why we have a discussion going to decided on the API :)